### ISSUE TITLE
Can't run segment anything

### ISSUE THREAD

**Issue by H68s12** [Uminosachi/sd-webui-inpaint-anything#72]
2023-07-09 05:44:30,912 - Inpaint Anything - INFO - input_image: (838, 672, 3) uint8
2023-07-09 05:44:31,675 - Inpaint Anything - INFO - SamAutomaticMaskGenerator sam_vit_b_01ec64.pth
2023-07-09 05:44:32,432 - Inpaint Anything - ERROR - The size of tensor a (0) must match the size of tensor b (256) at non-singleton dimension 1

**Comment by Uminosachi** [comment]
The issue does not reproduce. It may be due to an outdated torch package. Please delete the `venv` folder in stable-diffusion-webui, restart `webui.bat` or `webui.sh`, and reinstall the package.

**Comment by H68s12** [comment]
I did that but the still the same error. Any other ideas what might be causing it? I am using and AMD gpu.

**Comment by Uminosachi** [comment]
I haven't had the opportunity to test this with AMD's GPU. Nonetheless, you may consider running it on your CPU by executing the following command:

```bash
webui.bat --use-cpu all --no-half --no-half-vae
```

Furthermore, there is a guide in the WebUI wiki that provides instructions on how to run the program with AMD GPUs. You may find this information helpful:

https://github.com/AUTOMATIC1111/stable-diffusion-webui/wiki/Install-and-Run-on-AMD-GPUs#running-natively

**Comment by Gui205** [comment]
I have the same issue, do you found an solution without use the CPU  ?

**Comment by Gui205** [comment]
> I haven't had the opportunity to test this with AMD's GPU. Nonetheless, you may consider running it on your CPU by executing the following command:
> 
> ```shell
> webui.bat --use-cpu all --no-half --no-half-vae
> ```
> 
> Furthermore, there is a guide in the WebUI wiki that provides instructions on how to run the program with AMD GPUs. You may find this information helpful:
> 
> https://github.com/AUTOMATIC1111/stable-diffusion-webui/wiki/Install-and-Run-on-AMD-GPUs#running-natively

I don't understand where i have to put this command ?

**Comment by Uminosachi** [comment]
> I don't understand where i have to put this command ?

You can edit the `webui-user.bat` file directly and set the arguments for `COMMANDLINE_ARGS` to control the command line arguments. Here's how you can do it:

In the `webui-user.bat` file, add or modify the following line:
```batch
set COMMANDLINE_ARGS=--use-cpu all --no-half --no-half-vae
```
Save the file, and then you can run it by double-clicking the `webui-user.bat` file or executing it from the command line.

**Comment by Gui205** [comment]
> > I don't understand where i have to put this command ?
> 
> You can edit the `webui-user.bat` file directly and set the arguments for `COMMANDLINE_ARGS` to control the command line arguments. Here's how you can do it:
> 
> In the `webui-user.bat` file, add or modify the following line:
> 
> ```batchfile
> set COMMANDLINE_ARGS=--use-cpu all --no-half --no-half-vae
> ```
> 
> Save the file, and then you can run it by double-clicking the `webui-user.bat` file or executing it from the command line.

Oh ok i see it's in user, i miss understood with the webui.

**Comment by cheremo** [comment]
conclusion for amd users: force the use of the CPU or buy a Nvidia card? :(
i regret buying the 7900xtx

**Comment by poweredmert** [comment]
i don't understand how this can work on cpu easily but cannot work on amd gpus. the fact that you said "haven't had the opportunity to test this with AMD's GPU" shows how terrible it is to have an amd gpu for this hobby

**Comment by Cyanus000** [comment]
> i don't understand how this can work on cpu easily but cannot work on amd gpus. the fact that you said "haven't had the opportunity to test this with AMD's GPU" shows how terrible it is to have an amd gpu for this hobby

yes, amd gpu has too many bugs when using dml on windows, there is another bug in tiled diffusion that stuck me for days

> conclusion for amd users: force the use of the CPU or buy a Nvidia card? :( i regret buying the 7900xtx

when i bought amd gpu, i only tended to play games, but later when i started to use stable diffusion, i realised it was a terrible choice. so many bugs and the VRAM utilization is too low

**Comment by sasoriza** [comment]
> > conclusion for amd users: force the use of the CPU or buy a Nvidia card? :( i regret buying the 7900xtx
> 
> when i bought amd gpu, i only tended to play games, but later when i started to use stable diffusion, i realised it was a terrible choice. so many bugs and the VRAM utilization is too low

Same :(
